test/unit/error.spec.ts

```diff
@@ -292,3 +285,109 @@ test("return from nested finally", () => {
     `;
     expect(util.transpileAndExecute(code)).toBe("finally AB");
 });
+
+test.each([`"Hello error string!"`, `42`, `true`, `false`, `undefined`, `{ x: "Hello error object!" }`])(
+    "throw and catch custom error object",
+    error => {
+        util.testFunction`
+            try {
+                throw ${error};
+            } catch (error) {
+                return error;
+            }
+        `.expectToMatchJsResult();
+    }
+);
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError"])(
+    "throw builtin Errors as classes",
+    errorType => {
+        util.testFunction`
+            try {
+                throw new ${errorType}("message")
+            } catch (error) {
+                if (error instanceof Error) {
+                    return error.toString();
+                } else {
+                    throw Error("This error serves to prevent false positives from .expectNoExecutionError()");
+                }
+            }
+        `.expectToMatchJsResult();
+    }
+);
+
+test.each(["Error", "RangeError", "ReferenceError", "SyntaxError", "TypeError", "URIError"])(
+    "throw builtin Errors as functions",
```

Comment:
`throwing as functions` doesn't sound as a good description. Actually IMO it would be better to test throwing anything and `Error` builtin separately.

So `throw and catch custom error object` would actually test that throw (and catch) pass anything as it is, and builtin class features are tested without any throws (i.e. just `return ${errorType}("message").toString()` there)